plugins/mimecast/komand_mimecast/util/api.py

```diff
@@ -179,17 +179,33 @@ def _handle_zip_file(self, request: requests.request) -> List[Dict]:
             List[Dict]: A list of dictionaries, where each dictionary represents a log entry extracted from the ZIP file
         """
         try:
-            content = self.sanitise_content(request.content)
-            with ZipFile(BytesIO(content)) as my_zip:
-                combined_json_list = [
-                    log for file_name in my_zip.namelist() for log in json.loads(my_zip.read(file_name)).get("data")
-                ]
+            with ZipFile(BytesIO(request.content)) as my_zip:
```

Comment:
I've updated this further to include as much debug error logs and to continue if we hit a bad file.